build/lib/locFunc.ts

```diff
@@ -24,33 +29,141 @@ export function packageLangpacksStream(): NodeJS.ReadWriteStream {
 		})
 
 	const builtLangpacks = langpackDescriptions.map(langpack => {
-		return fromLocalNormal(langpack.path)
+		return ext.fromLocalNormal(langpack.path)
 			.pipe(rename(p => p.dirname = `langpacks/${langpack.name}/${p.dirname}`));
 	});
 
 	return es.merge(builtLangpacks);
 }
 
-//copied from extensions.
-function fromLocalNormal(extensionPath: string): Stream {
-	const result = es.through();
+export function packageADSExtensionsStream(): NodeJS.ReadWriteStream {
+	const currentADSJson = JSON.parse(fs.readFileSync(path.join(__dirname, '../../i18n/ADSExtensions.json'), 'utf8'));
+	const ADSExtensions = currentADSJson.ADSExtensions;
+	const extenalExtensionDescriptions = (<string[]>glob.sync('extensions/*/package.json'))
+		.map(manifestPath => {
+			const extensionPath = path.dirname(path.join(root, manifestPath));
+			const extensionName = path.basename(extensionPath);
+			return { name: extensionName, path: extensionPath };
+		})
+		.filter(({ name }) => ADSExtensions[name] !== undefined);
 
-	const vsce = require('vsce') as typeof import('vsce');
+	const builtExtensions = extenalExtensionDescriptions.map(extension => {
+		return ext.fromLocal(extension.path, false)
+			.pipe(rename(p => p.dirname = `extensions/${extension.name}/${p.dirname}`));
+	});
 
-	vsce.listFiles({ cwd: extensionPath, packageManager: vsce.PackageManager.Yarn })
-		.then(fileNames => {
-			const files = fileNames
-				.map(fileName => path.join(extensionPath, fileName))
-				.map(filePath => new File({
-					path: filePath,
-					stat: fs.statSync(filePath),
-					base: extensionPath,
-					contents: fs.createReadStream(filePath) as any
-				}));
+	return es.merge(builtExtensions);
+}
 
-			es.readArray(files).pipe(result);
-		})
-		.catch(err => result.emit('error', err));
+export function createXlfFilesForExtensions(): ThroughStream {
```

Comment:
It looks like this exists already in https://github.com/microsoft/azuredatastudio/blob/main/build/lib/i18n.ts - is there a reason you're duplicating it here?

Reply:
I will export that instead.